src/data/actions/assets.test.jsx

```diff
@@ -159,15 +206,72 @@ describe('Assets Action Creators', () => {
     expect(store.dispatch(actionCreators.uploadExceedMaxCount(1))).toEqual(expectedAction);
   });
   it('returns expected state from uploadAssetFailure', () => {
-    const expectedAction = { file: 'file', response: 'response', type: assetActions.UPLOAD_ASSET_FAILURE };
+    const expectedAction = { asset: 'file', response: 'response', type: assetActions.UPLOAD_ASSET_FAILURE };
     expect(store.dispatch(actionCreators.uploadAssetFailure('file', 'response'))).toEqual(expectedAction);
   });
   it('returns expected state from uploadAssetSuccess', () => {
-    const expectedAction = { asset: 'asset', response: 'response', type: assetActions.UPLOAD_ASSET_SUCCESS };
-    expect(store.dispatch(actionCreators.uploadAssetSuccess('asset', 'response'))).toEqual(expectedAction);
+    const expectedAction = { response: 'response', type: assetActions.UPLOAD_ASSET_SUCCESS };
+    expect(store.dispatch(actionCreators.uploadAssetSuccess('response'))).toEqual(expectedAction);
   });
   it('returns expected state from uploadingAssets', () => {
     const expectedAction = { count: 99, type: assetActions.UPLOADING_ASSETS };
     expect(store.dispatch(actionCreators.uploadingAssets(99, 'response'))).toEqual(expectedAction);
   });
+  it('returns expected state from uploadAssets success', () => {
+    const assetsResponse = {
+      status: 200,
+      body: {
+        assets: ['a.txt'],
+      },
+    };
+
+    fetchMock.mock(`begin:${assetsEndpoint}`, getUploadResponse(true), { method: 'post' });
```

Comment:
getUploadResponse is a currying function, which, given a boolean that represents whether the mocked response represents a success or a failure, returns a function that fetch-mock uses to return a mocked response. This allows me to dynamically mock a response that has the correct filename for each file.